packages/main/src/TableSelection.ts

```diff
@@ -204,6 +217,181 @@ class TableSelection extends UI5Element implements ITableFeature {
 		this._table.headerRow[0]._invalidate++;
 		this._table.rows.forEach(row => row._invalidate++);
 	}
+
+	_onkeydown(e: KeyboardEvent) {
+		if (!this.isMultiSelect() || !this._table || !e.shiftKey) {
+			return;
+		}
+
+		const focusedElement = getActiveElement(); // Assumption: The focused element is always the "next" row after navigation.
+
+		if (focusedElement === null) {
+			this._stopRangeSelection();
+			return;
+		}
+
+		if (!(isInstanceOfTableRow(focusedElement)) && !this._rangeSelection?.isMouse) {
+			this._stopRangeSelection();
+			return;
+		}
+
+		if (!this._rangeSelection) {
+			// If no range selection is active, start one
+			this._startRangeSelection(focusedElement as TableRow);
+		} else if (e.key === "ArrowUp" || e.key === "ArrowDown") {
+			const change = isUpShift(e) ? -1 : 1;
+			this._handleRangeSelection(focusedElement as TableRow, change);
+		}
+	}
+
+	_onkeyup(e: KeyboardEvent, eventTarget: HTMLElement) {
```

Comment:
eventTarget is a little misleading i question then why not we access e.target. Can we rename it eventOrigin?